stix2/registration.py

```diff
@@ -3,10 +3,83 @@
 from . import registry
 from .base import _DomainObject, _Observable
 from .exceptions import DuplicateRegistrationError
-from .properties import _validate_type
-from .utils import PREFIX_21_REGEX, get_class_hierarchy_names
+from .properties import (
+    ListProperty, ObjectReferenceProperty, ReferenceProperty, _validate_type,
+)
+from .utils import PREFIX_21_REGEX
 from .version import DEFAULT_VERSION
 
+# Properties ending in "_ref/s" need to be instances of specific types to meet
+# our interpretation of section 3.1 of the spec
+_VERSION_REF_TYPES_MAP = {
+    # TODO: The 2.0 code seems to use `ReferenceProperty`s - is that okay?
+    "2.0": (ObjectReferenceProperty, ReferenceProperty),
+    "2.1": (ReferenceProperty,),
+}
```

Comment:
Would it be reasonable to make this public (remove the underscore) and turn these values from `tuple`s into `set`s (or something similarly mutable) so that a downstream developer like me could do something like `VERSION_REF_TYPES_MAP["2.1"].add(MyCustomReferenceType)`?